apps/web/app/(app)/environments/[environmentId]/settings/api-keys/actions.ts

```diff
@@ -1,11 +1,30 @@
 "use server";
 
-import { deleteApiKey, createApiKey } from "@formbricks/lib/services/apiKey";
+import { authOptions } from "@/app/api/auth/[...nextauth]/authOptions";
+import { hasUserEnvironmentAccess } from "@/lib/api/apiHelper";
+import { deleteApiKey, createApiKey, canUserAccessApiKey } from "@formbricks/lib/services/apiKey";
 import { TApiKeyCreateInput } from "@formbricks/types/v1/apiKeys";
+import { Session, getServerSession } from "next-auth";
 
 export async function deleteApiKeyAction(id: string) {
-  return await deleteApiKey(id);
+  const session: Session | null = await getServerSession(authOptions);
```

Comment:
should we also add a `if (!session) {throw ...}` here?

And do you need the explicit type of session? Isn't `getServerSession` giving this already?